sound/soc/sof/ipc3-pcm.c

```diff
@@ -139,8 +139,8 @@ static int sof_ipc3_pcm_hw_params(struct snd_soc_component *component,
 	return ret;
 }
 
-static int sof_ipc3_pcm_trigger(struct snd_soc_component *component,
-				struct snd_pcm_substream *substream, int cmd)
+static int sof_ipc3_pcm_pre_trigger(struct snd_soc_component *component,
+				    struct snd_pcm_substream *substream, int cmd)
```

Comment:
commit message: I've seen all these explanations before, and we've already upstreamed code to fix things related to ordering already. It appears they were not good enough, so for the second pass I think it's required to explain what the shortcomings of the previous solution were, and why this one is better.